pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -151,15 +152,47 @@ type DNSProxy struct {
 	rejectReply int32
 }
 
+type FqdnsEntry struct {
+	fqdns   map[string]struct{}
+	fqdnKey *string
+}
+
+type RegexpAndFqdnsEntry struct {
+	regexp *regexp.Regexp
+	fqdns  *FqdnsEntry
+}
+
+type regexpIndexEntry struct {
+	regexp  *regexp.Regexp
+	counter int
+}
+
+type fqdnsIndexEntry struct {
+	fqdns   *FqdnsEntry
+	counter int
+}
+
 // perEPAllow maps EndpointIDs to ports + selectors + rules
-type perEPAllow map[uint64]portToSelectorAllow
+type perEPAllow struct {
+	allowMap    map[uint64]portToSelectorAllow
+	regexpIndex map[string]*regexpIndexEntry
+	fqdnsIndex  map[string]*fqdnsIndexEntry
```

Comment:
It would be good to document what we expect these keys that are strings to be

Reply:
Yee, did a small cleanup now and added some comment. Feel free to take a look!